Gems/ONNX/Code/Source/Clients/Model.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Debug/Timer.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <ONNX/ONNXBus.h>
+#include <onnxruntime_cxx_api.h>
+
+#include "upng/upng.h"
+
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <string>
+
+namespace ONNX
+{
+    //! Generic ONNX model which can be used to create an inference session and run inferences.
+    class Model
+    {
+    public:
+        Model()
+        {
+        }
+        //! Required params to create session and run inference, passed to Load().
+        struct InitSettings
+        {
+            //! Source of onnx model file.
+            std::wstring m_modelFile = std::wstring{ W_GEM_ASSETS_PATH } + std::wstring{ L"/model.onnx" };
```

Comment:
Let's use AZStd::string here

Reply:
Needs to be a wstring due to Ort::Session requiring the filepath to be as such. Changed to AZStd::wstring.

Reply:
Actually AZ::IO::Path should be used here.
Furthermore the modelFile path should be initialized in a constructor to get access to the `AZ::IO::FileIOBase` instance in order to resolve the path to the gem.

```c++
InitSettings()
{
    AZ::IO::FileIOBase* fileIo = AZ::IO::FileIOBase::GetInstance();
    
    if (AZ::IO::FixedMaxPath defaultModelPath;
        fileIo != nullptr && fileIo->ResolvePath(defaultModelPath, "@gemroot:ONNX@/Assets/model.onnx"))
    {
        m_modelFile = defaultModelPath.String();
    }
}
AZ::IO::Path m_modelFile;
```

Reply:
Also were is the wstring API at
I don't see it in the Ort::Session [API](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_session.html)